File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        for (int i=0; i< strs[0].size(); i++){
            char x = strs[0][i];
            for(int j=1; j< strs.size(); j++){
                if(x != strs[j][i]){
                    return ans;
                }
            }
            ans+=x;
        }
    return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        x = 0
        ans = ""
        if len(strs) == 1 or strs[0] == "":
            return strs[0]
            
        while x < len(strs[0]) + 1:
            if strs[0][:x] != strs[len(strs) - 1][:x]:
                return strs[0][:x-1]
            x += 1
        return strs[0][:x]
